### PR TITLE
fix and simplify workspace exposure strategies doc

### PR DESCRIPTION
**Pull request by sparkoo**
Signed-off-by: Michal Vala <mvala@redhat.com>

> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
update workspace exposure strategies page, simplify it and focus more on simple installation steps

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18498
https://github.com/eclipse/che/issues/18453

### Specify the version of the product this PR applies to.
now

### PR Checklist

As the author of this Pull Request I made sure that:

- [ ] `vale` has been run successfully against the PR branch
- [ ] Link checker has been run successfully against the PR branch
- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [ ] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

**Review by yhontyk — APPROVED** [review]
LGTM

**Comment by sparkoo** [comment]
I've added network policy requirement warning for gateway single-host, fixing this issue https://github.com/eclipse/che/issues/18453

File: modules/installation-guide/partials/con_workspace-exposure-strategies.adoc [hunk 1]
```diff
@@ -19,23 +19,29 @@ ifeval::["{project-context}" == "che"]
...
-When any of those single-host types is used, all workspaces are deployed to sub-paths of the main {prod-short} server domain.
+With single-host strategy, all workspaces are deployed to sub-paths of the main {prod-short} server domain.
 
 This is convenient for TLS-secured {prod-short} servers because it is sufficient to have a single certificate for the {prod-short} server, which will cover all the workspace component deployments as well.
 
+Single-host strategy have two subtypes with different implementation methods. First subtype is named `native`. This strategy is available and default on {kubernetes}, but not on OpenShift, since it uses Ingresses for servers exposing. The second subtype named `gateway`, works both on {platforms-name}, and uses a special pod with reverse-proxy running inside to route requests.
+
+[WARNING]
+====
+With `gateway` single-host strategy, cluster network policies has to be to be configured so that workspace's services are reachable from reverse-proxy Pod (typically in {prod-short} {orch-namespace}). These typically lives in different {orch-namespace}.
```
Comment (skabashnyuk) [inline modules/installation-guide/partials/con_workspace-exposure-strategies.adoc]: `has to be to be` copy&paste error

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 2]
```diff
@@ -13,7 +13,17 @@ link:https://docs.openshift.com/container-platform/latest/operators/olm-what-ope
...
+
+The supported values for `spec.server.serverExposureStrategy` are:
+
+* xref:multi-host-workspace-exposure-strategy_{context}[`multi-host`]
+* xref:single-host-workspace-exposure-strategy_{context}[`single-host`]
+ifeval::["{project-context}" == "che"]
+* xref:default-host-workspace-exposure-strategy_{context}[`default-host`]
+endif::[]
+
+See http://0.0.0.0:4000/che-7/installation-guide/configuring-workspace-exposure-strategies/#workspace-exposure-strategies_configuring-workspace-exposure-strategies for more detail about individual strategies.
```
Comment (skabashnyuk) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]: is this link legitimate?

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 3]
```diff
@@ -41,14 +51,28 @@ metadata:
...
+  "path": "/spec/server/serverExposureStrategy",
+  "value": "__<exposure-strategy>__"}]' \ <1>
+  -n __<namespace-name>__
+----
+<1> - used workspace exposure strategy
+
+ifeval::["{project-context}" == "che"]
+
+.Gateway single-host on Kubernetes
+Single-host on Kubernetes has 2 implementations, `native`(default) and `gateway`. To deploy with `gateway` set the `k8s.singleHostExposureType` of CheCluster Custom Resource to `gateway`, or use this patch:
```
Comment (skabashnyuk) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]: has 2 implementations ->  has two implementations

**Review by skabashnyuk — APPROVED** [review]
looks good in general.

File: modules/installation-guide/partials/con_workspace-exposure-strategies.adoc (hunk 1 above)
Comment (sparkoo) [reply modules/installation-guide/partials/con_workspace-exposure-strategies.adoc]: done

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc (hunk 2 above)
Comment (sparkoo) [reply modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]: fixed

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc (hunk 3 above)
Comment (sparkoo) [reply modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]: fixed

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 4]
```diff
@@ -58,30 +82,9 @@ metadata:
 spec:
   server:
     serverExposureStrategy: 'single-host'
-  k8s:
```
Comment (sleshchenko) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc:61]: It's not a valid yaml, is it? Is it done by purpose?

Comment (sparkoo) [reply modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc:61]: nice catch. fixed

**Pull request merged**